tests/e2e/specs/backend/product-query/product-filters.test.ts

```diff
@@ -0,0 +1,273 @@
+/**
+ * External dependencies
+ */
+import { canvas, setPostContent, insertBlock } from '@wordpress/e2e-test-utils';
+import {
+	visitBlockPage,
+	saveOrPublish,
+	selectBlockByName,
+	findToolsPanelWithTitle,
+	getFixtureProductsData,
+	getFormElementIdByLabel,
+	shopper,
+	getToggleIdByLabel,
+} from '@woocommerce/blocks-test-utils';
+import { ElementHandle } from 'puppeteer';
+import { setCheckbox, unsetCheckbox } from '@woocommerce/e2e-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	GUTENBERG_EDITOR_CONTEXT,
+	describeOrSkip,
+	waitForCanvas,
+	openBlockEditorSettings,
+} from '../../../utils';
+
+const block = {
+	name: 'Product Query',
+	slug: 'core/query',
+	class: '.wp-block-query',
+};
+
+/**
+ * Selectors used for interacting with the block in the editor. These selectors
+ * can be changed upstream in Gutenberg, so we scope them here for
+ * maintainability.
+ *
+ * There are also some labels that are used repeatedly, but we don't scope them
+ * in favor of readability. Unlike selectors, those label are visible to end
+ * users, so it's easier to understand what's going on if we don't scope them.
+ * Those labels can get upated in the future, but the tests will fail and we'll
+ * know to update them.
+ */
+const SELECTORS = {
+	productFiltersDropdownButton: {
+		default:
+			'.components-tools-panel-header .components-dropdown-menu button[aria-expanded="false"]',
+		expanded:
+			'.components-tools-panel-header .components-dropdown-menu button[aria-expanded="true"]',
+	},
```

Comment:
This is just a suggestion, but a pattern I found useful in my previous work as QA Engineer: using something called a “selector factory”. Basically, instead of a string, using a function to return a selector. This would look like so:

```suggestion
	productFiltersDropdownButton:
	  ( expanded: boolean = false ) => `.components-tools-panel-header .components-dropdown-menu button[aria-expanded="${ expanded }"]`,
```

You avoid repetition and you can use it like so: `canvas().click( productFiltersDropdownButton( true ) )`. If you want to be even more explicit (which I don't mind) you can do:

```
	productFiltersDropdownButton:
	  ( { expanded }: { expanded: boolean } = { expanded: false } ) => `.components-tools-panel-header .components-dropdown-menu button[aria-expanded="${ expanded }"]`,
```

Definition is a bit more verbose, but usage is a bit more explicit: `canvas().click( productFiltersDropdownButton( { expanded: true } ) )`.

These are just suggestions for patterns I found useful in the past.